cargo-guppy/src/lib.rs

```diff
@@ -18,18 +33,180 @@ pub fn cmd_diff(json: bool, old: &str, new: &str) -> Result<(), Error> {
     Ok(())
 }
 
-pub fn cmd_count() -> Result<(), Error> {
+pub fn cmd_count() -> Result<(), anyhow::Error> {
     let lockfile = Lockfile::from_file("Cargo.lock")?;
 
     println!("Third-party Packages: {}", lockfile.third_party_packages());
 
     Ok(())
 }
 
-pub fn cmd_dups() -> Result<(), Error> {
+pub fn cmd_dups() -> Result<(), anyhow::Error> {
     let lockfile = Lockfile::from_file("Cargo.lock")?;
 
     lockfile.duplicate_packages();
 
     Ok(())
 }
+
+arg_enum! {
+    #[derive(Debug)]
+    pub enum Kind {
+        All,
+        Workspace,
+        ThirdParty,
+    }
+}
+
+struct NameVisitor;
+
+impl PackageDotVisitor for NameVisitor {
+    fn visit_package(&self, package: &PackageMetadata, mut f: DotWrite<'_, '_>) -> fmt::Result {
+        write!(f, "{}", package.name())
+    }
+
+    fn visit_link(&self, _link: DependencyLink<'_>, mut f: DotWrite<'_, '_>) -> fmt::Result {
+        write!(f, "")
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct SelectOptions {
+    #[structopt(flatten)]
+    filter_opts: FilterOptions,
+
+    #[structopt(long, rename_all = "kebab-case")]
+    /// Save selection graph in .dot format
+    output_dot: Option<String>,
+
+    #[structopt(rename_all = "screaming_snake_case")]
+    /// The root packages to start the selection from
+    roots: Vec<String>,
```

Comment:
I added a big note about. Since the expectation is that the root set would be your workspace members these won't conflict with anything. Bare package names were used to avoid having to do a lot of typing of workspace paths.